x-pack/plugins/observability_solution/dataset_quality/server/routes/data_streams/routes.ts

```diff
@@ -106,31 +106,39 @@ const dataStreamDetailsRoute = createDatasetQualityServerRoute({
     path: t.type({
       dataStream: t.string,
     }),
+    query: rangeRt,
   }),
   options: {
     tags: [],
   },
   async handler(resources): Promise<DataStreamDetails> {
-    const { context, params } = resources;
+    const { context, params, getEsCapabilities } = resources;
     const { dataStream } = params.path;
+    const { start, end } = params.query;
     const coreContext = await context.core;
 
     // Query datastreams as the current user as the Kibana internal user may not have all the required permissions
     const esClient = coreContext.elasticsearch.client.asCurrentUser;
 
-    const [type, ...datasetQuery] = dataStream.split('-');
+    const { type, dataset, namespace } = indexNameToDataStreamParts(dataStream);
+    const sizeStatsAvailable = !(await getEsCapabilities()).serverless;
 
     const [dataStreamsStats, dataStreamDetails] = await Promise.all([
       getDataStreamsStats({
         esClient,
-        type: type as DataStreamType,
-        datasetQuery: datasetQuery.join('-'),
+        type,
+        datasetQuery: `${dataset}-${namespace}`,
       }),
-      getDataStreamDetails({ esClient, dataStream }),
+      getDataStreamDetails({ esClient, dataStream, start, end, sizeStatsAvailable }),
     ]);
 
     return {
       createdOn: dataStreamDetails?.createdOn,
+      docsCount: dataStreamDetails?.docsCount,
```

Comment:
[nit] I would create a new route for dataset summary, here we are querying `createdOn` and `lastActivity` everytime that time range is changed, imo those two values do not depend on the timerange

Reply:
Created a new route `/settings` to separate the `createdOn` out which shouldn't be refreshed every time the time range changes.

The existing route `/details` could be renamed to `/summary`, but I guess both work for this use case so I've left it as is to `/details`.

`lastActivity ` is still part of `/details` as it should be refreshed along with the time range.

Reply:
> The existing route /details could be renamed to /summary, but I guess both work for this use case so I've left it as is to /details.

up to you.

Thanks for the other changes